### DISCUSSION TITLE
Installing with Container Manager on Synology NAS, so many issues

### DISCUSSION THREAD

**Discussion by metaclam** [haugene/docker-transmission-openvpn#2838] [Q&A]
I am a newbie with containers and VPNs and pretty much everything involved, but I managed to get a container running transmission with ghrc.io/linuxserver/transmission, very well.  I'll call that "basic".  (There I followed a guide which involved running scripts.)

I've completely failed with haugene and I hardly know where to start!

• Should I start with a Project?  Or Creating a Container?  They have similar results.  The former said the container was running within the project and that isn't what basic was doing --- so last attempt was creating a container directly.

• In trying to edit the settings of the container there are a number of fields -- like WEBPROXY_PORT that seem auto-added but have no values.  Container Manager won't let me save other changes until I either filll in those values or remove them.

• removing them, the container starts but immediately crashes.  how to troubleshoot?

• the default settings don't include things like LOCAL_NETWORK that are in this project's .yml example files.  Should I add them?  (I also tried starting the project with a .yml file that _did_ have that key, but had similar problems)

• Please explain LOCAL_NETWORK.  mine is in the range of 192.168.0.1 - 192.168.0.255.  Should I put 192.168.0.1/255 ?  What's the 192.168.0.0/24 I see in examples?   Why end with 0?  .1 is the router.

Here's the [.json settings file](https://pastebin.com/zwu7MJwR) on pastebin*.  And here's the [log file](https://pastebin.com/JGkU72nD) from Container Manager for the container.

(I just subscribed to PIA and have a username / password.  All 000 in the pastebin is dummy data, rest is actual)

*edited new attempt 2024-05-16

**Comment by pkishino** [comment]
I’ll let others comment but please read through the documentation for the container first, second I don’t think using dsm built in Manager will work well.. perhaps instead of using the cli you can setup portainer first and then setup this container through that.

**Reply by metaclam** [reply]
sigh.  I have read thru documentation but much of it implies knowledge of systems I've never used.  I'm comfortable with working with command line interfaces and SSH and such so I'm not completely inexperienced, but basically I need clear step-by-step directions that apply to a specific platform.

I think many are in the same boat, Synology seems to be a common NAS, it has a Container Management app -- yet no one is writing good documentation specific to that.  I have heard of Portainer but never seen it let alone used it or know how to set it up on Synology.  I understand that's hardly your job -- and thank you kindly for your work -- but wish someone would write tailored instructions.

I'm not sure why Container Manager shouldn't work given the right settings to plugin there (and the right ones to remove).  It also has an option to start with a .yml file, and I tried that unsuccessfully too (starting with the example given in this project).

The LOCAL_NETWORK setting I asked about is a great example.  I know how to configure my LAN on my router, I know basically what the range is for DHCP, but have no idea how to express that in a setting, such as what "/24" is, as described above.

If there are issues with the settings file I linked to I'd be pleased to know about them (and I updated these based on info in other threads, with same results).  TYVM.

**Reply by metaclam** [reply]
FYI -- I just tried it with SSH logged into an admin user which I want to be the user with the PUID/GUID, using the docker run command:

 ```
docker run --cap-add=NET_ADMIN -d \
>               -v /docker/trm-data:/data \
>               -v /docker/trm-home:/config \
>               -e OPENVPN_PROVIDER=PIA \
>               -e OPENVPN_CONFIG=us_west \
>               -e OPENVPN_USERNAME=u00000 \
>               -e OPENVPN_PASSWORD=pw00000 \
>               -e LOCAL_NETWORK=192.168.0.0/16 \
>               --log-driver json-file \
>               --log-opt max-size=10m \
>               -p 9091:9091 \
>               haugene/transmission-openvpn
docker: Got permission denied while trying to connect to the Docker daemon socket at unix:///var/run/docker.sock: Post "http://%2Fvar%2Frun%2Fdocker.sock/v1.24/containers/create": dial unix /var/run/docker.sock: connect: permission denied.
```

would it work if I ran as sudo?  I also don't trust this with basic stuff -- like how does it know what to create in the docker file system (other than data & config)?  in other methods I created files and folders manually.  this has no target at all.

edit: yes -- it does install with sudo.  and I borrowed a --name=containername flag after -d from another source and that worked.

(but still back to the same problems as using Container Manager.  It adds the same extra fields, and cannot continue until they are removed or filled in.  but it does start at least.)

The the file structure is different than the "basic" transmission install, with the main torrent & resume files going in the subdirectory of transmission-home.  

What goes in the root directory for /config then?  and what is /data used then? confusing.

**Comment by metaclam** [comment]
so.  consider this thread my learning process.  creating it with docker run worked. (and it's not obvious to someone like me that you can SSH into the Synology box -- which in turn means turning on the SSH service, and execute that "docker run" command as root with sudo).  

the thing starts.  editing the settings with Container Manager also works, apart from the annoying bit about those auto-populated keys needing values.

I still don't get the LOCAL_NETWORK variable*, and I cannot connect with a browser, [even after reading this](https://haugene.github.io/docker-transmission-openvpn/vpn-networking/#access_the_webui).  I added the / to /transmission/rpc/ .  I am running it on port 9092 because Container Manager complained that 9091 was in use -- probably by that "basic" transmission container install (which is not running).

edit: OK figured out LOCAL_NETWORK and [what /24 means](https://www.oreilly.com/library/view/virtual-private-networks/1565925297/ch01s05.html) -- it's the whole range to 255.  however, still working on actually connecting.

**Comment by pkishino** [comment]
The  LOCAL_NETWORK is specify your subnet.. if your devices are all on 192.168.0.1-256 range then you could put 192.168.0.0/24 here.. else you can use a subnet calculator to check/ BRPatrick KishinoOn 17 May 2024, at 00:54, metaclam ***@***.***> wrote:﻿
so.  consider this thread my learning process.  creating it with docker run worked. (and it's not obvious to someone like me that you can SSH into the Synology box -- which in turn means turning on the SSH service, and execute that "docker run" command as root with sudo).
the thing starts.  editing the settings with Container Manager also works, apart from the annoying bit about those auto-populated keys needing values.
I still don't get the LOCAL_NETWORK variable, and I cannot connect with a browser, even after reading this.  I added the / to /transmission/rpc/ .  I am running it on port 9092 because Container Manager complained that 9091 was in use -- probably by that "basic" transmission container install.

**Reply by metaclam** [reply]
thanks! and I'm in!  not obvious to me!

the network connection errors had to do with binding port 9092 but still having the RPC port at 9091 in the settings.

challenges now include these:
• does port forwarding work?  can I access the WebUI from the WAN?

• one private tracker says "invalid port" for all its torrents but no others do. Why?

• how to make various random settings stick -- settings that are available in different WebUI and Transmission Remote GUI inconsistently.  settings to remove torrents when added or to add torrents stopped seem not to stick -- either when starting and stopping transmisison or when switching between say, Flood (in the internal package) and Trans. Remote GUI.   Not clear to me where all these settings files are kept.

• Flood shows error in Transmission.log about not being able to read its conflg file (file doesn't exist) in /opt/ ; not sure which /opt, relative to the container or to the NAS?  

• how do those WebUIs get updated?  (for example, I have a reason ATM to keep very up to date with Shift)

• how does the container get updated, for that matter?  does doing so destroy any settings?

• seems to be a bug in this container where if you set TRANSMISSION_WATCH_DIR to an explicit path relative to the volume mounts, it ignores that in favor of /data/watch.   I removed that key and bound /watch to an actual folder in docker and that worked.

• no doubt much much more ;)  

Thanks for your work, man.
